app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,13 +78,140 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
                 dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
             }
+            binding.pauseSwitch.setOnClickListener {
```

Comment:
So much code duplication.
Similar code is found in NewProfileFragment.
Move the similar code to a single object file and call the functions using that object file only.

Reply:
@rob729 I have moved the duplicate functions in Utlis.kt, please check.